### ISSUE TITLE
State-based <input> tag renders incorrectly, possible problem with virtual dom diffing

### ISSUE THREAD

**Issue by klick-barakgall** [yysun/apprun#2]
I have the following view function:
```
  view = (state) => {
    return <div>
      <h1>Welcome to the app!</h1>
      <form id="login" action="api/login.php" method="POST" onsubmit={(ev) => {
        this.run('login:submit', ev);
        return false;
      }}>
        <input type="email" name="email" placeholder="your@email.address" required />
        {state === 'login:signup' ? <input type="text" name="name" placeholder="Your name" required /> : ''}
        <input type="submit" />
      </form>
    </div>;
  }
```

Basically it's a form with a single input unless `state === 'login:signup'` where by it will have two inputs (email and name).

This is working fine:
 * On initial load I only see the email input
 * When state changes to `login:signup` I see both email and name inputs

However, when state reverts back to anything other then `login:signup` the name input remains!

I know virtual dom diffing algorithms have lots of trouble dealing with inputs. I wonder if this might be an issue there?

**Comment by yysun** [comment]
To hide the element, use a space instead of empty string.
```
{state === 'login:signup' ? <input /> : ' ' }  // works
{state === 'login:signup' ? <input /> : '' }   // not work
```

**Comment by yysun** [comment]
Works with null too
```
{state === 'login:signup' ? <input /> : null }  // works
```

Actually it is a bug in virtual dom. It has been fixed in 1.7.3.

**Comment by yysun** [comment]
It has been fixed in v1.7.5.
